### PR TITLE
Include aliases when searching the dex

### PR DESCRIPTION
**Pull request by urkerab**
I was wondering why my `/dt oricorio-balie` was failing. It turns out that we don't search aliases, so it couldn't tell me that I had mistyped `Oricorio-Baile`.

In order to search aliases I changed the code to search the dex keys; hopefully this doesn't change anything significantly.

### CODE REVIEW

**Comment by Zarel** [comment]
I have no clue what this code is doing. Can someone else review?

**Comment by urkerab** [comment]
How the old code worked:

- Each search type is looked up to see if there is an exact match. If there is, then the exact item is returned.
- Otherwise, each search type is scanned for an inexact match.
  - The amount of inexactness depends on the length of the search string.
- The best (i.e. least inexact) match is then selected.
- If this is not the original term, then the result is recursively obtained.

How my code works:

- Each search item is looked up to see if there is an exact match. If there is, then the exact item is returned.
- Otherwise, if this was a recursive call, then the search fails.
- The search terms and aliases are then checked for an inexact match.
  - The amount of inexactness initially depends on the length of the search string.
  - The inexactness is reduced as better matches are found, thus resulting in the best match.

Miscellaneous code differences:

- The match is now checked to ensure that it's relevant to the current generation (for future use).
- If the match is inexact, the name of the inexact match is used (this makes a difference when it's an alias).
- Inexact matching now works on ids, so that Aliases can be compared.

File: sim/dex.js [hunk 1]
```diff
@@ -933,68 +933,51 @@ class ModdedDex {
...
-				let ld = this.levenshtein(cmpTarget, word.toLowerCase(), maxLd);
+				let ld = this.levenshtein(cmpTarget, j, maxLd);
 				if (ld <= maxLd) {
-					fuzzyResults.push({word: word, ld: ld});
-				}
-			}
-		}
-
-		if (fuzzyResults.length) {
-			/** @type {any} */
```
Comment (Zarel) [inline sim/dex.js]: ?

Comment (Zarel) [inline sim/dex.js]: Oh nvm that was the old code

**Comment by Zarel** [comment]
I still don't really understand this code, but it doesn't look any worse than the old code... :/

**Pull request merged**